### ISSUE TITLE
Documentation Update Needed for Role Assumption

### ISSUE THREAD

**Issue by northrup** [kubernetes-sigs/cluster-api-provider-aws#1964]
The documentation for Role Assumption appears to be out of date and referencing example files and templates that no longer exist in the repo. It is incredibly hard to follow what is happening or what should be happening between CAPA and KIAM.

**Comment by randomvariable** [comment]
I'm probably going to remove the "setting up kiam" bits, as that's out of scope, and will quickly get outdated.

**Comment by randomvariable** [comment]
/close

superseded by #2053
